llvm/lib/Target/AArch64/MCTargetDesc/AArch64ELFObjectWriter.cpp

```diff
@@ -211,18 +211,18 @@ unsigned AArch64ELFObjectWriter::getRelocType(MCContext &Ctx,
               Target.getAccessVariant() == MCSymbolRefExpr::VK_GOTPCREL)
                  ? ELF::R_AARCH64_GOTPCREL32
                  : R_CLS(ABS32);
-    case FK_Data_8:
+    case FK_Data_8: {
+      bool IsAuth = (RefKind == AArch64MCExpr::VK_AUTH ||
+                     RefKind == AArch64MCExpr::VK_AUTHADDR);
       if (IsILP32) {
         Ctx.reportError(Fixup.getLoc(),
-                        "ILP32 8 byte absolute data "
-                        "relocation not supported (LP64 eqv: ABS64)");
+                        Twine("ILP32 8 byte absolute data "
```

Comment:
8-byte absolute data relocation is not supported in ILP32?

Reply:
According to the spec, it's not - see no "ELF32 code" for `R_<CLS>_ABS64` in the table here https://github.com/ARM-software/abi-aa/blob/main/aaelf64/aaelf64.rst#static-data-relocations

Reply:
Ok, I was suggesting a candidate diagnostic message

`8-byte absolute data relocation is not supported in ILP32`

Reply:
It looks like that error messages across the function use some convention, so it's probably better to stick with it (or change the messages everywhere at once)